app/presenters/monitoring-stations/view-monitoring-stations.presenter.js

```diff
@@ -0,0 +1,135 @@
+    stationReference: monitoringStation.stationReference,
+    wiskiId: monitoringStation.wiskiId
+  }
+}
+
+function _alertedUpdatedAt (licenceDetails) {
+  if (licenceDetails.statusUpdatedAt) {
+    return formatLongDate(licenceDetails.statusUpdatedAt)
+  }
+
+  return formatLongDate(licenceDetails.createdAt)
+}
+
+function _alertType (licence) {
+  if (licence.alertType === 'stop') {
+    return 'Stop'
+  }
+
+  if (licence.alertType === 'reduce') {
+    return 'Reduce'
+  }
+
+  return 'Stop or reduce'
+}
+
+function _checkPermissions (auth, roleType) {
+  return auth.credentials.roles.some((role) => {
+    return role.role === roleType
+  })
+}
+
+function _createPageTitle (riverName, stationName) {
+  if (riverName) {
+    return `${riverName} at ${stationName}`
+  }
+
+  return stationName
+}
+
+function _formatLicenceDetailsAbstractionPeriod (licenceDetails) {
+  return formatAbstractionPeriod(
+    licenceDetails.abstractionPeriodStartDay,
+    licenceDetails.abstractionPeriodStartMonth,
+    licenceDetails.abstractionPeriodEndDay,
+    licenceDetails.abstractionPeriodEndMonth
+  )
+}
+
+function _formatLicences (licenceDetails) {
+  return licenceDetails.map((licenceDetail) => {
+    return {
+      abstractionPeriod: _formatLicenceDetailsAbstractionPeriod(licenceDetail),
+      alertType: _alertType(licenceDetail),
+      alertUpdatedAt: _alertedUpdatedAt(licenceDetail),
+      createdAt: licenceDetail.createdAt,
+      lastUpdatedAt: licenceDetail.statusUpdatedAt,
+      id: licenceDetail.licence.id,
+      licenceRef: licenceDetail.licence.licenceRef,
+      restrictionType: licenceDetail.restrictionType === 'flow' ? 'Flow' : 'Level',
+      threshold: `${licenceDetail.thresholdValue} ${licenceDetail.thresholdUnit}`
+    }
+  })
+}
+
+function _groupLicences (licences) {
+  // NOTE: This function groups licence objects by their unique `id`. It takes the array of licences and uses the
+  // `reduce` method to accumulate an object, where each key is a licence `id`. For each unique `id`, a new object is
+  // created with `licenceId`, `licenceRef`, and an empty `linkages` array. The current licence object is then pushed
+  // into the `linkages` array of the corresponding group. Finally, the grouped licences are returned as an array of
+  // these grouped objects.
+  const groupedLicences = licences.reduce((grouped, current) => {
+    const { id, licenceRef } = current
+
+    if (!grouped[id]) {
+      grouped[id] = {
+        licenceId: id,
+        licenceRef,
+        linkages: []
+      }
+    }
+
+    grouped[id].linkages.push(current)
+
+    return grouped
+  }, {})
+
+  return Object.values(groupedLicences)
+}
+
+function _sortLicences (licences) {
+  return licences.sort((licenceA, licenceB) => {
+    if (licenceA.licenceRef !== licenceB.licenceRef) {
+      return licenceA.licenceRef < licenceB.licenceRef ? -1 : 1
+    }
+
+    return 0
+  })
+}
```

Comment:
Are we not able to sort the data in the fetch query rather than here in code? If struggling get in contact and we can look together.